### PR TITLE
checkpoints: Show how to manage checkpoints

### PR DESCRIPTION
**Pull request by nirs**
Replace remove_checkpint example with checkpoints example, supporting
sub commands for managing checkpoints.

The script provides 2 sub commands:

- remove - remove specific checkpoint (same as previous
  remove_checkpoint)

- purge - remove checkpoint older than 7 days. The numebr of days is
  configurable using the --days argument.

The new purge option is useful for developers or users using
backup_vm.py, since this script never remove checkpoints. It is also
useful for users using backup applications to clean up after the backup
application in case the application does not remove old checkpoints.

### CODE REVIEW

**Comment by nirs** [comment]
@vjuranek @bennyz please review.

File: examples/checkpoints.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+            checkpoint_age = now - checkpoint.creation_date
+
+            if checkpoint_age.days > args.days:
+                progress(f"Removing checkpoint {checkpoint.id}, created {checkpoint_age.days} ago")
+                checkpoint_service = checkpoints_service.checkpoint_service(checkpoint.id)
+
+                # Removing the checkpoint
+                checkpoint_service.remove()
+                try:
+                    while checkpoint_service.get():
```
Comment (vjuranek) [inline examples/checkpoints.py:128]: Same here.

File: examples/checkpoints.py [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        now = datetime.datetime.now(datetime.timezone.utc)
+
+        for checkpoint in checkpoints_service.list():
+            checkpoint_age = now - checkpoint.creation_date
+
+            if checkpoint_age.days > args.days:
+                progress(f"Removing checkpoint {checkpoint.id}, created {checkpoint_age.days} ago")
+                checkpoint_service = checkpoints_service.checkpoint_service(checkpoint.id)
+
+                # Removing the checkpoint
```
Comment (vjuranek) [inline examples/checkpoints.py:125]: As we wait for a checkpoint to be removed, maybe run it in a thread so the removal run in parallel? Also if removing one checkpoint fails for some reason of get stuck, it won't affect removal of other checkpoints.

**Review by vjuranek — CHANGES_REQUESTED** [review]

File: examples/checkpoints.py [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+        # Validate that the VM has the requested checkpoint
+        try:
+            checkpoint_service.get()
+        except sdk.NotFoundError:
+            raise RuntimeError(f"VM {args.vm_uuid} has no checkpoint {args.checkpoint_uuid}")
+
+        # Removing the checkpoint
+        checkpoint_service.remove()
+        try:
+            while checkpoint_service.get():
```
[the remark this answers is not in the document]
Comment (nirs) [reply examples/checkpoints.py:100]: Yes, but we did not have one before - this patch just add another command. We can improve this later.

File: examples/checkpoints.py (hunk 2 above)
Comment (nirs) [reply examples/checkpoints.py:125]: It is possible to do the wait in parallel, but removing a checkpoint is very quick, so there is no need to make the example more complicated.

File: examples/checkpoints.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (vjuranek) [reply examples/checkpoints.py:100]: ack

File: examples/checkpoints.py (hunk 2 above)
Comment (vjuranek) [reply examples/checkpoints.py:125]: makes sense

**Pull request merged**